File: backend/src/main/java/com/portnet/etrade/demand/entity/Demand.java
package com.portnet.etrade.demand.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "demands")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Demand {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String reference;

    @Column(name = "exporter_id", nullable = false)
    private String exporterId;

    @Column(name = "exporter_name", nullable = false)
    private String exporterName;

    @Column(nullable = false)
    private String currency;

    @Column(nullable = false, precision = 15, scale = 2)
    private BigDecimal amount;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private DemandStatus status;

    @Column(name = "bill_of_lading")
    private String billOfLading;

    @Column(name = "invoice_number")
    private String invoiceNumber;

    @Column
    private String description;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    public enum DemandStatus {
        DRAFT, SUBMITTED, UNDER_REVIEW, APPROVED, REJECTED, FINANCED
    }
}

File: frontend/src/types/demand.ts
export enum DemandStatus {
  DRAFT = 'DRAFT',
  SUBMITTED = 'SUBMITTED',
  UNDER_REVIEW = 'UNDER_REVIEW',
  APPROVED = 'APPROVED',
  REJECTED = 'REJECTED',
  CANCELLED = 'CANCELLED',
}

export interface Demand {
  id: string;
  reference: string;
  exportatorId?: string;
  type: string;
  amount: number;
  currency: string;
  status: DemandStatus;
  description?: string;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateDemandDto {
  type: string;
  amount: number;
  currency: string;
  description?: string;
}

export interface UpdateDemandDto extends Partial<CreateDemandDto> {
  status?: DemandStatus;
}
